RousseauBot.py

```diff
@@ -1,64 +1,76 @@
+try:
+    with open('prevIndices.txt', 'rb') as indexFile:
+        prevIndex = pk.load(indexFile)
+except:
+    prevIndex = []
 
-# search for indices where the label for each credential is stored
-apiKeyInd, apiSecretInd, accTokenInd, accTokSecretInd = \
-    credList.index('API Key'), credList.index('API Secret Key'), credList.index('Access Token'), credList.index('Access Token Secret')
-# then get each credential as the succeeding item to each label above
-apiKey, apiSecret, accessToken, accessTokSecret = \
-    credList[apiKeyInd + 1], credList[apiSecretInd + 1], credList[accTokenInd + 1], credList[accTokSecretInd + 1]
+# Generate a random number such as we won't index a line from the corpus that was already indexed in the past 10 days
+index = np.random.randint(0, len(corpus)-1)
+while index in prevIndex:
+    index = np.random.randint(0, len(corpus)-1)
+    
+# Store current index into the previous indices list to be pickled, for keeping track going forward
+prevIndex.append(index)
 
-# read in the corpus file, containing Rousseau's "On the Social Contract" broken out into lines
-with open('corpus.txt', 'r') as file:
-    corpus = file.readlines()
-# read in a pickled list, keeping track of previous lines from the corpus that was quoted in the past 10 days
-# this helps prevent the possibility of repeating the same line within 10 days
-with open('prevIndices.txt', 'rb') as indFile:
-    prevInd = pk.load(indFile)
+# Just want to keep track over 10 days
+if len(prevIndex) > 10:
+    prevIndex.pop(0)
+    
+# Pickle the previous indices list
+with open('prevIndices.txt', 'wb') as indexFile:
+    pk.dump(prevIndex, indexFile)
 
-# generate a random number such as we won't index a line from the corpus that was already indexed in the past 10 days
-ind = np.random.randint(0, len(corpus)-1)
-while ind in prevInd:
-    ind = np.random.randint(0, len(corpus)-1)
-# store current index into the previous indices list to be pickled, for keeping track going forward
-prevInd.append(ind)
-# just want to keep track over 10 days
-if len(prevInd)>10:
-    prevInd.pop(0)
-# pickle the previous indices list
-with open('prevIndices.txt', 'wb') as indFile:
-    pk.dump(prevInd, indFile)
+quoteToTweet = corpus[index].strip('\n')
 
-quoteToTweet = corpus[ind].strip('\n')
+# Authenticate credentials with Twitter to access RousseauBot's account
+try:
+    auth = tw.OAuthHandler(apiKey, apiSecret)
+except:
+    sys.exit("API request rejected by Twitter.")
 
-# authenticate credentials with Twitter to access RousseauBot's account
-auth = tw.OAuthHandler(apiKey, apiSecret)
-auth.set_access_token(accessToken, accessTokSecret)
+try:
+    auth.set_access_token(accessToken, accessTokSecret)
+except:
+    sys.exit("Authentication failed.")
+    
 api = tw.API(auth)
 
-# tweet on RousseauBot's account
-api.update_status(quoteToTweet)
\ No newline at end of file
+# Tweet on RousseauBot's account
+api.update_status(quoteToTweet)
```

Comment:
Should you be sending a tweet if either of the previous 2 calls result in an exception?

Reply:
With the sys.exit in the previous two excepts though, would this last line still be executed if the previous calls (Auth/Access) resulted in an error?

Reply:
shoot, my bad! You're right, the `sys.exit` calls take care of them.